src/map.cpp

```diff
@@ -1327,6 +1327,28 @@ ter_id map::ter(const int x, const int y) const
     return current_submap->get_ter( lx, ly );
 }
 
+bool map::pl_knows( const tripoint &p ) const
+{
+    if( !INBOUNDS(p.x, p.y) ) {
+        return false;
+    }
+
+    int lx, ly;
+    submap * const current_submap = get_submap_at(p, lx, ly);
+    return current_submap->known[lx][ly];
+}
+
+void map::set_pl_known( const tripoint &p )
+{
+    if( !INBOUNDS(p.x, p.y) ) {
```

Comment:
What about `p.z`? There is now a `inbounds` function that accepts a single tripoint, use that instead.